### PR TITLE
initial pipeline template for angular frontend

### PR DESCRIPTION
**Pull request by scott-robertson1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-37

### Change description ###
initial jenkinsfiles for darts-portal

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!groovy
+
+@Library("Infrastructure")
+
+def type = "angular"
+def product = "darts"
+def component = "darts-portal"
```
Comment (davet1985) [inline Jenkinsfile_CNP:7]: Not sure if this should be?

```suggestion
def product = "darts"
def component = "portal"
```

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+properties([
+// H allow predefined but random minute see https://en.wikipedia.org/wiki/Cron#Non-standard_characters
+pipelineTriggers([cron('H 05 * * *')])
+])
+
+@Library("Infrastructure")
+
+def type = "angular"
+def product = "rpe"
```
Comment (jackmaloney) [inline Jenkinsfile_nightly:11]: Shouldn't this be "darts"?

File: Jenkinsfile_nightly [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+def component = "darts-portal"
```
Comment (jackmaloney) [inline Jenkinsfile_nightly:12]: And then this should be "portal"?

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+def yarnBuilder = new uk.gov.hmcts.contino.YarnBuilder(this)
+
+withPipeline(type, product, component) {
+  afterAlways('build') {
+    yarnBuilder.yarn('rebuild puppeteer')
+    yarnBuilder.yarn('build')
+  }
```
Comment (davet1985) [inline Jenkinsfile_CNP:15]: Not sure we want to run puppeteer right now...

**Review by davet1985 — APPROVED** [review]
LGTM

**Pull request merged**